crates/rolldown/src/stages/generate_stage/code_splitting.rs

```diff
@@ -408,11 +408,85 @@ impl GenerateStage<'_> {
         continue;
       }
 
-      if let Some(allow_min_size) = match_groups[this_module_group.match_group_index]
+      let allow_min_size = match_groups[this_module_group.match_group_index]
         .min_size
         .map_or(chunking_options.min_size, Some)
+        .unwrap_or(0.0);
+
+      if this_module_group.sizes < allow_min_size {
+        continue;
+      }
+
+      if let Some(allow_max_size) = match_groups[this_module_group.match_group_index]
+        .max_size
+        .map_or(chunking_options.max_size, Some)
       {
-        if this_module_group.sizes < allow_min_size {
+        if this_module_group.sizes > allow_max_size {
+          // If the size of the group is larger than the max size, we should split the group into smaller groups.
+          let mut modules = this_module_group.modules.iter().copied().collect::<Vec<_>>();
+          modules.sort_by_key(|module_idx| {
+            self.link_output.module_table.modules[*module_idx].exec_order()
```

Comment:
These three sorts could be merged into one.

Reply:
Done